src/Markdig/Syntax/Block.cs

```diff
@@ -48,6 +50,12 @@ protected Block(BlockParser parser)
         /// </summary>
         public bool RemoveAfterProcessInlines { get; set; }
 
+        public StringSlice BeforeWhitespace { get; set; }
+        public StringSlice AfterWhitespace { get; set; }
+
+        public List<StringSlice> LinesBefore { get; set; }
+        public List<StringSlice> LinesAfter { get; internal set; }
```

Comment:
I would expect that StringSlice are a better fit when working with an existing document, because you don't have to create multiple substrings of the original document to fill all the little whitespace blocks that we would have.

But when creating a new document from the API, each StringSlice would most likely be instantiated from plain separated strings.

Reply:
Indeed. StringSlice has a ctor accepting a String. If such apis will be used in the future, we may (carefully) consider an implicit operator from string, so api user can assign strings to these fields.

Is it safe to say we should use StringSlice types as general guideline for these fields primarily used for roundtrip parsing/generating?

Reply:
I have now changed the current PR into using StringSlice everywhere where possible. Please let me know if this will resolve this item.